File: src/Goods.BackOffice/client-app/src/domain/drivers/driverBlank.ts
import { DriverLicenseCategory } from "./enums/driverLicenseCategory";
import { Gender } from "./enums/gender";
import { Driver } from "./driver";

export interface DriverBlank {
  id: string | null;
  firstName: string | null;
  secondName: string | null;
  lastName: string | null;
  gender: Gender | null;
  driverLicenseCategory: DriverLicenseCategory[];
  birthday: Date | null;
  experience: Date | null;
  payPerHour: number | null;
}

export namespace DriverBlank {
  export function getDefault(): DriverBlank {
    return {
      id: null,
      firstName: null,
      secondName: null,
      lastName: null,
      gender: null,
      driverLicenseCategory: [],
      birthday: null,
      experience: null,
      payPerHour: null
    };
  }

  export function fromDriver(driver: Driver): DriverBlank {
    return {
      id: driver.id,
      firstName: driver.firstName,
      secondName: driver.secondName,
      lastName: driver.lastName,
      gender: driver.gender,
      driverLicenseCategory: driver.driverLicenseCategory,
      birthday: driver.birthday,
      experience: driver.experience,
      payPerHour: driver.payPerHour
    }
  }
}

File: src/Goods.Domain/Drivers/DriverBlank.cs
using Goods.Domain.Drivers.Enums;

namespace Goods.Domain.Drivers;

public class DriverBlank
{
    public Guid? Id { get; set; }
    public String? FirstName { get; set; }
    public String? SecondName { get; set; }
    public String? LastName { get; set; }
    public Gender? Gender { get; set; }
    public LicenseCategory[]? DriverLicenseCategory { get; set; }
    public DateOnly? Birthday { get; set; }
    public DateOnly? Experience { get; set; }
    public Decimal? PayPerHour { get; set; }
    public DateTime? CreatedDatetimeUTC { get; set; }
    public DateTime ModifiedDatetimeUTC { get; set; }
    public Boolean IsRemoved { get; set; }
}
